fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/models/FedoraResource.java

```diff
@@ -242,4 +242,11 @@ void replaceProperties(final IdentifierConverter<Resource, FedoraResource> idTra
      * @return the node for this object at the version provided
      */
     Node getNodeVersion(String label);
+
+    /**
+     * This method returns the version label of this frozen resource.
+     * If this resource is not frozen, null is returned.
+     * @return version label
+     */
+    String getVersionLabelOfFrozenResource();
```

Comment:
:+1: to giving the `kernel-api` a thorough review from the perspective of modeling commitments... as a follow-on task.